url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/syntax/builder/SyntaxBuilderParserTables.java

```diff
@@ -0,0 +1,720 @@
+
+		syntax.addRule(s = new Rule("quantifiedsequnit", 2));	// rule 17
+		s.addRightSymbol("sequnit");
+		s.addRightSymbol("`quantifier`");
+
+		syntax.addRule(s = new Rule("quantifiedsequnit", 1));	// rule 18
+		s.addRightSymbol("sequnit");
+
+		syntax.addRule(s = new Rule("sequence", 2));	// rule 19
+		s.addRightSymbol("sequence");
+		s.addRightSymbol("quantifiedsequnit");
+
+		syntax.addRule(s = new Rule("sequence", 1));	// rule 20
+		s.addRightSymbol("quantifiedsequnit");
+
+		syntax.addRule(s = new Rule("sequence_opt", 1));	// rule 21
+		s.addRightSymbol("sequence");
+
+		syntax.addRule(s = new Rule("sequence_opt", 0));	// rule 22
+
+		syntax.addRule(s = new Rule("unionseq", 3));	// rule 23
+		s.addRightSymbol("unionseq");
+		s.addRightSymbol("'|'");
+		s.addRightSymbol("sequence_opt");
+
+		syntax.addRule(s = new Rule("unionseq", 1));	// rule 24
+		s.addRightSymbol("sequence_opt");
+
+		syntax.addRule(s = new Rule("rule", 4));	// rule 25
+		s.addRightSymbol("`identifier`");
+		s.addRightSymbol("\"::=\"");
+		s.addRightSymbol("unionseq");
+		s.addRightSymbol("';'");
+
+
+		loadGotoTable();
+		loadParseActionTable();
+
+		terminalsWithoutEpsilon = new ArrayList(12);
+		terminalsWithoutEpsilon.add("`bnf_chardef`");
+		terminalsWithoutEpsilon.add("\"..\"");
+		terminalsWithoutEpsilon.add("`identifier`");
+		terminalsWithoutEpsilon.add("`ruleref`");
+		terminalsWithoutEpsilon.add("`stringdef`");
+		terminalsWithoutEpsilon.add("'-'");
+		terminalsWithoutEpsilon.add("'('");
+		terminalsWithoutEpsilon.add("')'");
+		terminalsWithoutEpsilon.add("`quantifier`");
+		terminalsWithoutEpsilon.add("'|'");
+		terminalsWithoutEpsilon.add("\"::=\"");
+		terminalsWithoutEpsilon.add("';'");
+	}
+	private void loadGotoTable()	{
+		gotoTable = new ArrayList(35);
+
+		loadGoto_0();
+		loadGoto_1();
+		gotoTable.add(null);  // state 2
+		loadGoto_3();
+		loadGoto_4();
+		gotoTable.add(null);  // state 5
+		loadGoto_6();
+		loadGoto_7();
+		gotoTable.add(null);  // state 8
+		loadGoto_9();
+		loadGoto_10();
+		gotoTable.add(null);  // state 11
+		gotoTable.add(null);  // state 12
+		loadGoto_13();
+		gotoTable.add(null);  // state 14
+		loadGoto_15();
+		gotoTable.add(null);  // state 16
+		gotoTable.add(null);  // state 17
+		gotoTable.add(null);  // state 18
+		gotoTable.add(null);  // state 19
+		gotoTable.add(null);  // state 20
+		loadGoto_21();
+		loadGoto_22();
+		gotoTable.add(null);  // state 23
+		gotoTable.add(null);  // state 24
+		loadGoto_25();
+		loadGoto_26();
+		gotoTable.add(null);  // state 27
+		loadGoto_28();
+		gotoTable.add(null);  // state 29
+		gotoTable.add(null);  // state 30
+		gotoTable.add(null);  // state 31
+		gotoTable.add(null);  // state 32
+		gotoTable.add(null);  // state 33
+		gotoTable.add(null);  // state 34
+	}
+	private void loadGoto_0()	{
+		Hashtable g = new Hashtable(3, 1);
+		gotoTable.add(g);
+		g.put("syntax", new Integer(3));
+		g.put("`identifier`", new Integer(1));
+		g.put("rule", new Integer(2));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 169:173, 236:240, 457:461, 645:649.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.